File: go/bob/bob.go
// Package bob solves Bob
package bob

import "regexp"

var (
	loud     = regexp.MustCompile("^[^[:lower:]]+[[:upper:]][^[:lower:]]+$")
	question = regexp.MustCompile("\\?[[:space:]]*$")
	shh      = regexp.MustCompile("^[[:space:]]*$")
)

// Hey responds to a remark.
func Hey(remark string) string {
	if loud.MatchString(remark) && question.MatchString(remark) {
		return "Calm down, I know what I'm doing!"
	} else if question.MatchString(remark) {
		return "Sure."
	} else if loud.MatchString(remark) {
		return "Whoa, chill out!"
	} else if shh.MatchString(remark) {
		return "Fine. Be that way!"
	} else {
		return "Whatever."
	}
}

File: python/bob/bob.py
def response(hey):
  if not hey or hey.isspace():
    return 'Fine. Be that way!'
  if hey.strip().endswith('?'):
    return "Calm down, I know what I'm doing!" if hey.isupper() else 'Sure.'
  else:
    return 'Whoa, chill out!' if hey.isupper() else 'Whatever.'

# vim:ts=2:sw=2:expandtab
